ctapipe/calib/camera/calibrator.py

```diff
@@ -109,9 +109,21 @@ def _calibrate_dl0(self, event, telid):
     def _calibrate_dl1(self, event, telid):
         waveforms = event.dl0.tel[telid].waveform
         selected_gain_channel = event.r1.tel[telid].selected_gain_channel
+        dl1_calib = event.calibration.tel[telid].dl1
+
         if self._check_dl0_empty(waveforms):
             return
+
         n_pixels, n_samples = waveforms.shape
+
+        # subtract any remaining pedestal before extraction
+        if dl1_calib.pedestal_offset is not None:
+            # this copies intentionally, we don't want to modify the dl0 data
+            # waveforms have shape (n_pixel, n_samples), pedestals (n_pixels, )
+            # so we need to transpose to be able to utilize broadcasting
+            # and transpose back after to retain the correct shape
+            waveforms = (waveforms.T - dl1_calib.pedestal_offset).T
```

Comment:
Is this procedure recommended over broadcasting with np.newaxis?

Reply:
It's not an alternative. It's different use cases, right?

I want to subtract from each row, but for broadcasting to work the last dimension has to be the same, so I have to transpose.

Reply:
Changed to the better readable solution with `[:, np.newaxis]`